internal/extract/pod_spec.go

```diff
@@ -47,3 +49,31 @@ func PodSpec(obj k8sutil.Object) (coreV1.PodSpec, bool) {
 	}
 	return podTemplateSpec.Spec, true
 }
+
+// Selector extracts a selector from the given object, if available.
+func Selector(obj k8sutil.Object) (*metaV1.LabelSelector, bool) {
```

Comment:
So, the second return value here is supposed to indicate whether the object is one that _has_ a selector (for example, it should return `false` if the object is a plain `Pod`). So I would remove the `hasSelector` thing from here, and just `return true` in every case where you're not explicitly returning `nil`.